### PR TITLE
Spreadsheets

### PR DESCRIPTION
**Pull request by vincentvankrunkelsven**
@richierocks Can you review this please? It probably needs another iteration and the part about Limitations needs more data once I finished the pivot tables functionality by the end of this week.

Note: there are some iframes in there that won't work on GitHub, but will work in the processed pages.

### CODE REVIEW

File: docs/spreadsheets/exercises/writing-scts.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  check = string,
+  diagnose = string
+)
+```
+
+This is the only **meta** SCT function. `check_correct` accepts 2 arguments: `check` and `diagnose`
+and both of them are SCTs themselves.
+
+The function will first check whether the SCT in `check` passes. If it passes, the SCT in `diagnose`
+is never executed and the SCT passes. If `check` fails, `diagnose` is ran and will generally give
```
Comment (richierocks) [inline docs/spreadsheets/exercises/writing-scts.md]: Can you recursively nest calls to `check_correct()`. As in

```
? check_correct(
  check = check_value,
  diagnose = {
    check_correct(
      check = check_function(function = "SUM")
      diagnose = check_reference
    );
  }
)
```

File: docs/spreadsheets/course-structure.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Course structure
```
[the remark this answers is not in the document]
Comment (richierocks) [reply docs/spreadsheets/course-structure.md]: Except the bit about workbooks. That needs to be documented.

File: docs/spreadsheets/workbooks.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+click on 'Add New Workbook' on the left-hand side. This will create a new workbook with one exercise
+scaffold in it.
+
+> #### warning::Warning
+>
+> Unfortunately there's currently no loader in the UI that shows the system is creating a
+> new workbook, however that doesn't mean the system isn't working. Before hitting the button again,
+> it's best to wait a few seconds and your new workbook should show up.
+
+## Golden copy
```
Comment (richierocks) [inline docs/spreadsheets/workbooks.md]: Maybe mention that this exists for performance reasons.

File: docs/spreadsheets/exercises/general.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+<iframe
+  src="https://docs.google.com/spreadsheets/d/e/2PACX-1vRi2lqbMZa5jjQgCcSNjyvDWg8v-NYuQ2cj9Vut-0PbUc2an2BIDk-q6R53f6zzAaFquH_MYV-Y2aKp/pubhtml?gid=790829783&single=true"
+  style="width:100%;height:385px;border:none"
+></iframe>
+
+A few things you can already notice here are:
+
+* An SCT is defined in a certain cell. This means the check will be done on this cell in the user's
+  solution. Specifically in our example, the value in `F2` will be checked.
+* An SCT starts with a question mark `? ...`. If there are multiple SCTS in a cell, they all start
```
[the remark this answers is not in the document]
Comment (vincentvankrunkelsven) [reply docs/spreadsheets/exercises/general.md]: - You can add zero to an infinite amount of spaces
- The space is not necessary. If you meant why the question mark is necessary? Two reasons: it's easier to differentiate between multiple SCTS like this & I found it useful to clearly see what parts in the SCT sheet are meant for testing and which parts are just there as a guidance to see where in the user's cell I'm placing the SCTs.

File: docs/spreadsheets/exercises/writing-scts.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+#### check_references
+
+```
+check_references(
+  message = string,
+  suggestion = string
+)
+```
+
+Similarly to `check_reference`, check whether the correct references are used in a cell. However,
```
[the remark this answers is not in the document]
Comment (vincentvankrunkelsven) [reply docs/spreadsheets/exercises/writing-scts.md]: - No, `check_reference` checks for the use of a reference, absolute references won't pass in your example, currently
- Yes, it works for references used inside function calls (clarified)

File: docs/spreadsheets/exercises/writing-scts.md (hunk 1 above)
Comment (vincentvankrunkelsven) [reply docs/spreadsheets/exercises/writing-scts.md]: Yes, you could use that. The syntax you proposed is slightly off though:
```
? check_correct(
  check = check_value,
  diagnose = {
    check_correct(
      check = check_function(function = "SUM")
      diagnose = check_reference
    );
  }
)
```

And you could simplify this to:

```
? check_correct(
  check = check_value,
  diagnose = check_correct(
    check = check_function(function = "SUM"),
    diagnose = check_references
  )
)
```

**Comment by vincentvankrunkelsven** [comment]
@richierocks I handled most of your remarks. Also added https://github.com/datacamp/spreadsheets/issues/154 and https://github.com/datacamp/spreadsheets/issues/153 for 2 of your remarks. Can I merge?

**Pull request merged**